### PR TITLE
feat: Add program provider to use in taxonomy-connector app to get program fields

### PR DESCRIPTION
**Pull request by AliAdnanSohail**
Ticket link: [3/5 - Create provider and validators for Program/Degree](https://2u-internal.atlassian.net/browse/PROD-2901)

## Description:
After adding provider and validator in taxonomy-connector app in [this PR](https://github.com/openedx/taxonomy-connector/pull/98) . We need to override there methods in discovery app to return Discovery Program data

### CODE REVIEW

**Review by ansabgillani — APPROVED** [review]

File: course_discovery/apps/taxonomy_support/providers.py [hunk 1]
```diff
@@ -53,3 +53,37 @@ def get_all_courses():  # lint-amnesty, pylint: disable=arguments-differ
...
+        """
+        Get iterator for all the programs (excluding drafts).
+        """
+        all_programs = Program.objects.all()
+        for chunked_programs in chunked_queryset(all_programs):
+            for program in chunked_programs:
+                yield {
+                    'uuid': program.uuid,
+                    'title': program.title,
+                    'subtitle': program.subtitle,
```
Comment (Ali-D-Akbar) [inline course_discovery/apps/taxonomy_support/providers.py:87]: Although it has been added in taxonomy already but do we need title and subtitle in this flow?

**Review by Ali-D-Akbar — APPROVED** [review]
Just a question otherwise, looks good.

File: course_discovery/apps/taxonomy_support/providers.py (hunk 1 above)
Comment (AliAdnanSohail) [reply course_discovery/apps/taxonomy_support/providers.py:87]: We internally discussed to pass title and subtitle to get skills of the Program

File: course_discovery/apps/taxonomy_support/providers.py [hunk 2]
```diff
@@ -53,3 +53,37 @@ def get_all_courses():  # lint-amnesty, pylint: disable=arguments-differ
...
+            'uuid': program.uuid,
+            'title': program.title,
+            'subtitle': program.subtitle,
+            'overview': program.overview,
+        } for program in programs]
+
+    @staticmethod
+    def get_all_programs():  # lint-amnesty, pylint: disable=arguments-differ
+        """
+        Get iterator for all the programs (excluding drafts).
```
Comment (DawoudSheraz) [inline course_discovery/apps/taxonomy_support/providers.py]: There is no draft concept in Programs.

File: course_discovery/apps/taxonomy_support/tests/test_providers.py [hunk 3]
```diff
@@ -22,17 +25,17 @@
...
+from course_discovery.apps.course_metadata.tests.factories import CourseFactory, ProgramFactory
 
 
 class TaxonomyIntegrationTests(TestCase):
     """
     Validate integration of taxonomy_support and metadata providers.
     """
 
-    def test_validate(self):
+    def test_validate_course_meta_data(self):
```
Comment (DawoudSheraz) [inline course_discovery/apps/taxonomy_support/tests/test_providers.py]: nit: course_metadata

**Review by DawoudSheraz — APPROVED** [review]
1-2 small nits

**Pull request merged**